py-polars/polars/internals/expr/expr.py

```diff
@@ -6296,6 +6299,7 @@ def inner_with_default(s: pli.Series) -> pli.Series:
             df = s.to_frame().unnest(s.name) if s.dtype == Struct else s.to_frame()
 
             column = df.columns[0]
+            input_dtype = df.select(column).dtypes[0]
```

Comment:
Maybe replace with: `input_dtype = df.dtypes[0]`

Reply:
Fixed in https://github.com/pola-rs/polars/pull/7217. I guess I didn't full realize that we always operate on the first field in the struct 😓